### ISSUE TITLE
Fair comparison- train/val/test

### ISSUE THREAD

**Issue by rahimentezari** [facebookresearch/open_lth#4]
Hi
As far as I see from the code, For each pruning ration, you report the test accuracy for the last epoch (#160). right?
How can I change the code to divide the training into train and validation? The goal is to report the test accuracy for the best validation accuracy.

**Comment by jfrankle** [comment]
You will need to modify the code to generate train, test, and validation sets (modifying the datasets module) and modify the corresponding test callback to evaluate accordingly.

**Comment by jfrankle** [comment]
Also, it records test accuracy for every epoch by default, although it only prints the last epoch on the command line.
